### DISCUSSION TITLE
How best to setup Saleor to work with headless CMS like Gatsby?

### DISCUSSION THREAD

**Discussion by amityweb** [saleor/saleor#9608] [Q&A]
I am new to headless CMSs, and Saleor. My client would like a new site with all their content on Strapi, then a static website setup for it all, but also eCommerce to sell products. 

So we thought a Strapi -> Gatsby / Saleor setup would be good. 

But what is the best way to add Saleor to a static site? And Gatsby for that matter? Doesn't have to be Gatsby, it's just one I chose. 

Would we have two websites, like www.mysite.com (Gatsby) and store.mysite.com (Saleor)? I prefer not to do this. 

Would all Saleor cart functions, like cart, checkout, mini-basket, customer login, orders, etc. all be pulled in from a Saleor backend behind the scenes? I guess some of this could be developed in Gatsby and Saleor not used (e.g. a cart) but customers and so their login and order history would all need to be fetched from Saleor. 

Where would the file system for Saleor be? Inside my Gatsby src, or a separate folder? When we publish Gastby it creates a /public folder, but we may also need to also fetch data from Saleor at that time. If we put everything in Gatsby I assume Saleor needs to be running somewhere to fetch all data (but we dont want it accessible to public), so I guess Saleor should not be inside Gatsby at all, but its own installation closed off from the public but accessible to Gatsby? 

Or do we even need Gatsby? Can Saleor be used as the headless CMS for any kind of static website data, not just store? 

Thanks for your patience!

**Comment by karolkielecki** [comment]
Hi @amityweb, you can create an integration (app) with an external headless CMS such as Strapi. You can find more info on how to do that here: https://docs.saleor.io/docs/3.x/developer/extending

**Comment by amityweb** [comment]
Thanks for the link. From further research I think it may be that we dont need Gatsby. Saleor Storefront is in Next.js so we could just create a website in that, then integrate the Saleor store features into it. 

I did install Saleor Storefront to see if that can be adapted by I cant believe how complex it is (perhaps its a Next.js thing). I have been building websites for over 20 years (using traditional/old school methods) and thought I would look into more modern approach, but its unbelievable how complex it is compared to traditional website building, which we can jump right in and start building websites instead of spending so much time on 3rd party services like docker. I have to even use docker on the production server by the looks of it. Its a shame to not be able to just deploy it to our web hosting servers, it would be easier and quicker. There is a massive learning curve (and issues that go wrong as I have already found) before we even start building the website. e.g. docker not working well. Ideally I would not want an extra services taking up server resources, when all I really need is the web hosting services installed normally, like PostgreSQL etc. I dont want to use Heroku, Netlify, etc. I find working with docker more complex than just installing a web server, PostgreSQL, Redit etc. Anyway, thats an off topic rant sorry!

I went to experiment changing the categories on the home page for example, I must have spent 3 or 4 hours until I found the category name in the .env file, but even that did not work. Looks like its outputting navigation content from Saleor rather than categories? Either way, working back from the index file that outputs the content to any query that specifies what data to fetch is really complex. 

I dont think documentation for editing Saleor Storefront is very good. I did not find any info on the categories used taken from an .env file, I had to figure it all out. 

So I thought it best I start building from scratch rather than adapting Saleor Storefront so I can create how it works how I see fit and understand it all, but then could be a lot more work building the eCommece side of it. And then the additional time and resources managing docker and any issues with that, not sure if its worth it all the extra effort, but will keep looking into it for a bit longer.

**Comment by tmechsner** [comment]
We recently integrated Wagtail CMS in a Saleor Shop that is built on top of the Saleor Storefront repo (started it just a few weeks before the discontinuation announcement... would suggest to start with the new "react-frontend" repo!). For cart, login, category pages etc. we use the storefront stuff. For content on product pages and for content pages we make calls to the Wagtail REST API from the React frontend. For integration with categories etc. in the CMS pages we work with manually entered slugs right now. Works great! Drop me an e-mail if you need further help. :)
